### PR TITLE
Aws sdk upgrade

### PR DESCRIPTION
**Pull request by jasonrichardsmith**
*Issue #, if available:*
#31 

*Description of changes:*
moved from glide to [go mod](https://github.com/Masterminds/glide#go-modules).
>The Go community is now using Go Modules to handle dependencies. Please consider using that instead of Glide. Glide is now mostly unmaintained.

Upgraded aws go sdk to newest for EKS OIDC IAM support.

Added docker build and push functionality. credentials still need to be added to the main repository.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/tracing/daemon.go [hunk 1]
```diff
@@ -20,20 +20,20 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/aws/aws-xray-daemon/daemon/bufferpool"
```
Comment (kiranmeduri) [inline cmd/tracing/daemon.go]: Is this change necessary for moving to go.mod? Can we scope it down non-code changes for faster rollout?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# build stage
+FROM golang:1.13-alpine AS build-env
```
Comment (kiranmeduri) [inline Dockerfile]: Thanks for adding this 👍

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+RUN adduser -D -u 10001 xray
+RUN CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-extldflags "-static"' \
+    -o daemon ./cmd/tracing/daemon.go ./cmd/tracing/tracing.go
+
+FROM scratch
+COPY --from=build-env /go/src/github.com/aws/aws-xray-daemon/daemon .
+COPY --from=build-env /etc/passwd /etc/passwd
+COPY --from=build-env /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+COPY pkg/cfg.yaml /etc/amazon/xray/cfg.yaml
+USER xray
```
Comment (kiranmeduri) [inline Dockerfile]: can you also add expose ports and fix the entrypoint according to https://docs.aws.amazon.com/xray/latest/devguide/xray-daemon-ecs.html.

Changing entrypoint may cause problems to folks who are overriding it.

File: Makefile [hunk 4]
```diff
@@ -11,13 +9,8 @@ endif
...
 export BGO_SPACE=$(shell pwd)
 path := $(BGO_SPACE):$(WORKSPACE)
-ifneq ($(GOPATH),)
-	GOPATH := $(path):$(GOPATH):$(BGO_SPACE)
-else
-	GOPATH := $(path):$(BGO_SPACE)
-endif
 
-export GOPATH
+# export GOPATH
```
Comment (kiranmeduri) [inline Makefile]: can this be dropped?

File: Makefile [hunk 5]
```diff
@@ -11,13 +9,8 @@ endif
 # Initialize BGO_SPACE
 export BGO_SPACE=$(shell pwd)
```
Comment (kiranmeduri) [inline Makefile]: is BGO_SPACE still needed?

File: cmd/tracing/daemon.go (hunk 1 above)
Comment (jasonrichardsmith) [reply cmd/tracing/daemon.go]: I have not changed any code.  I am just reorganizing the folders for best practices.  The only thing that has changed is the import directives.
https://github.com/golang-standards/project-layout#pkg

File: Makefile (hunk 5 above)
Comment (jasonrichardsmith) [reply Makefile]: I am fine leaving it for now.  It does not impact anything and I have seen projects that do this same thing.

File: Dockerfile (hunk 3 above)
Comment (jasonrichardsmith) [reply Dockerfile]: fixed

Comment (jasonrichardsmith) [reply Dockerfile]: done

File: Makefile (hunk 4 above)
Comment (jasonrichardsmith) [reply Makefile]: done

File: Dockerfile (hunk 2 above)
Comment (jasonrichardsmith) [reply Dockerfile]: :+1:

**Comment by jasonrichardsmith** [comment]
If this get merged, please manually push the latest tagged release using
```
make build-docker
make push-docker
```
I need to update this code to reflect a tagged image and add service accounts loading.

https://github.com/aws/aws-app-mesh-inject/blob/master/pkg/patch/sidecar.go#L74

**Comment by jasonrichardsmith** [comment]
Any chance for a review here?

**Comment by jasonrichardsmith** [comment]
I was wondering if we could get this reviewed.

**Comment by willarmiros** [comment]
Hi @jasonrichardsmith,
Thank you for your contribution. We will be reviewing it soon, unfortunately things get a little slower around the holidays.

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,14 @@
+module github.com/aws/aws-xray-daemon
+
+go 1.13
+
+require (
+	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
+	github.com/aws/aws-sdk-go v1.25.42
```
Comment (willarmiros) [inline go.mod]: Hello @jasonrichardsmith, 
Sorry for this PR not getting prioritized. We will be reviewing it as soon as we have bandwidth. In the meantime, could you please update the version of the `aws-sdk-go` module to update that in PR #42? I plan on closing that PR in favor of this one, but the upgrade must happen for internal testing issues.

**Comment by shengxil** [comment]
Merged at https://github.com/aws/aws-xray-daemon/commit/bb22d59e6c519f474fa489573cbf1c87f7fda13d. Closing this one
